### PR TITLE
telemetry: add telemetry for fast mode

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #35850

Problem Summary:

### What is changed and how it works?

Add telemetry（fast mode table proportion) for fast mode.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by hongyunyan** [comment]
@AstroProfundis @breezewish

File: telemetry/data_feature_usage.go [hunk 1]
```diff
@@ -279,3 +282,31 @@ func postReportNonTransactionalCounter() {
 func getGlobalKillUsageInfo() bool {
 	return config.GetGlobalConfig().EnableGlobalKill
 }
+
+// TiFlashModeStatistics records the usage info of Fast Mode
+type TiFlashModeStatistics struct {
+	// fast mode table / (fast mode table + normal mode table) , these tables must have tiflash replica
+	FastModeTableProportion float64 `json:"fast_mode_table_proportion"`
```
Comment (breezewish) [inline telemetry/data_feature_usage.go:289]: Prefer to just use raw data for larger data exploration potential, i.e. separated fields for number of fast mode tables, number of normal mode tables, number of all tables.

**Comment by hongyunyan** [comment]
run-check_dev_2-test

File: telemetry/data_feature_usage.go [hunk 2]
```diff
@@ -284,6 +287,32 @@ func getGlobalKillUsageInfo() bool {
 	return config.GetGlobalConfig().EnableGlobalKill
 }
 
+
+// TiFlashModeStatistics records the usage info of Fast Mode
+type TiFlashModeStatistics struct {
+	FastModeTableCount   int64 `json:"fast_mode_table_count"`
+	NormalModeTableCount int64 `json:"normal_mode_table_count"`
+}
```
Comment (breezewish) [inline telemetry/data_feature_usage.go:295]: How about also adding a field called `AllTableCount`?

**Review by breezewish — APPROVED** [review]
Generally LGTM

**Review by AstroProfundis — APPROVED** [review]

**Comment by hongyunyan** [comment]
build-unit-test

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
Thank you for your help on the linter.

**Pull request merged**